lib/connector.js

```diff
@@ -55,13 +61,25 @@ Connector.prototype.getMedadata = function () {
 
 /**
  * Execute a command with given parameters
- * @param {String} command The command such as SQL
- * @param {Object[]} [params] An array of parameters
+ * @param {String|Object} command The command such as SQL
+ * @param {*[]} [params] An array of parameter values
+ * @param {Object} [options] Options object
  * @param {Function} [callback] The callback function
  */
-Connector.prototype.execute = function (command, params, callback) {
+Connector.prototype.execute = function(command, params, options, callback) {
   /*jshint unused:false */
-  throw new Error('query method should be declared in connector');
+  throw new Error('execute() must be implemented by the connector');
+};
+Connector.prototype.command = Connector.prototype.query =
```

Comment:
If I am reading this code correctly, then if a connector implements `execute` method only, then the alias methods `command` and `query` will be still pointing to the base `execute` implementation which throws error.

I think a better solution is to define aliases via property getters:

```js
// a mockup, may not work OOTB
Object.defineProperty(Connector.prototype, 'command', { get: { return this.execute; } });
Object.defineProperty(Connector.prototype, 'query', { get: { return this.execute; } });
```